core/src/main/scala/org/apache/spark/util/collection/CompactBuffer.scala

```diff
@@ -152,6 +152,20 @@ private[spark] class CompactBuffer[T: ClassTag] extends Seq[T] with Serializable
 }
 
 private[spark] object CompactBuffer {
+  def constantEmpty[T: ClassTag](): CompactBuffer[T] = new CompactBuffer[T] {
```

Comment:
Is this really worth creating a new class for?  Creating new subclasses can actually affect how JIT is going to inline things so might end up degrading performance.